scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3927,604 @@ bool SlaOutputDev::checkClip()
+	qreal  lineWidth = 0.0;
+#ifdef DEBUG_TEXT_IMPORT
+	qDebug() << "rendering new frame at:" << xCoor << "," << yCoor << " With lineheight of: " << activeTextRegion->lineSpacing << "Height:" << activeTextRegion->maxHeight << " Width:" << activeTextRegion->maxWidth;
+#endif
+	/* colours don't get reset to CommonStrings::None often enough.*/
+	int z = m_doc->itemAdd(PageItem::TextFrame, PageItem::Rectangle, xCoor, yCoor, 40, 40, 0, CommonStrings::None, CommonStrings::None /* this->CurrColorStroke*/);//, PageItem::ItemKind::InlineItem);
+	PageItem* textNode = m_doc->Items->at(z);
+
+	ParagraphStyle& pStyle = (ParagraphStyle&)textNode->itemText.defaultStyle();
+	// set some hackish parameters up at first, line spacing can be calculated from the cursor position changes
+	pStyle.setLineSpacingMode(pStyle.AutomaticLineSpacing);
+	pStyle.setHyphenationMode(pStyle.AutomaticHyphenation);
+
+	// TODO: Implement these using the framework
+	finishItem(textNode);
+	// FIXME: Implement these using the framework
+	//_setFillAndStrokeForPdf(state, text_node);
+	//FIXME: Here's some dummy code for now with sensible defaults, looks like state wasn't even needed
+
+	textNode->ClipEdited = true;
+	textNode->FrameType = 3;
+	textNode->setLineEnd(PLineEnd);
+	textNode->setLineJoin(PLineJoin);
+	textNode->setTextFlowMode(PageItem::TextFlowDisabled);
+	//textNode->setFillTransparency(1.0);
+	textNode->setLineTransparency(1.0); // this sets the transparency of the textbox border and we don't want to see it			
+	textNode->setFillColor(CommonStrings::None);
+	textNode->setLineColor(CommonStrings::None);
+	textNode->setLineWidth(0);//line  width doesn't effect drawing text, it creates a bounding box state->getTransformedLineWidth());
+	textNode->setFillShade(CurrFillShade);
+
+
+	// Oliver Stieber 2020-06-11 Set text matrix... This need to be done so that the global world view that we rite out glyphs to is transformed correctly by the context matrix for each glyph, possibly anyhow.
+	// needs the way in whicvh we are handeling transformations for the page to be more concrete beofre this code can be implemented either here or somewhere else
+	/* FIXME: Setting the text matrix isn't supported at the moment 
+	QTransform text_transform(_text_matrix);
+	text_transform.setMatrix(text_transform.m11(), text_transform.m12(), 0,
+		text_transform.m21(), text_transform.m22(), 0,
+		first_glyph.position.x(), first_glyph.position.y(), 1);
+		*/
+	/* todo, set the global transform
+	gchar *transform = sp_svg_transform_write(text_transform);
+	text_node->setAttribute("transform", transform);
+	g_free(transform);
+	*/
+	/*set the default charstyle to the style of the glyph, this needs fleshing out a little */
+
+	int shade = 100;
+	//TODO: This needs to come from the framework

```

Comment:
you should use `PdfTextRecognition` instead of framework... somebody who was not involved in the discussion would not understand what the framework is.